go-controller/pkg/ovn/master.go

```diff
@@ -549,29 +549,81 @@ func (oc *Controller) SetupMaster(masterNodeName string) error {
-	stdout, stderr, err := util.RunOVNNbctl("--", "--may-exist", "lsp-add", logicalSwitch, portName,
-		"--", "lsp-set-type", portName, portType,
-		"--", "lsp-set-options", portName, options,
-		"--", "lsp-set-addresses", portName, addresses)
+func addNodeLogicalSwitchPort(ovnNBClient goovn.Client, logicalSwitch, portName, portType, addresses, options string) (string, error) {
+	var cmds []*goovn.OvnCommand
+	var goOvnOptions map[string]string
+
+	lsp, err := ovnNBClient.LSPGet(portName)
+	if err != nil && err != goovn.ErrorNotFound && err != goovn.ErrorSchema {
+		return "", fmt.Errorf("unable to get the lsp: %s from the nbdb: %s", portName, err)
+	}
+
+	//Add LSP if it does not exist
+	if lsp == nil {
+		lspAddCmd, err := ovnNBClient.LSPAdd(logicalSwitch, portName)
+		if err != nil {
+			return "", fmt.Errorf("failed to add logical port %s to switch %s error: %v", portName, logicalSwitch, err)
+		}
+		cmds = append(cmds, lspAddCmd)
+	}
+	cmd, err := ovnNBClient.LSPSetAddress(portName, addresses)
+	if err != nil {
+		return "", fmt.Errorf("failed to set address %s to port %s error: %v", addresses, portName, err)
+	}
+	cmds = append(cmds, cmd)
+
+	if options != "" {
+		goOvnOptions = parseLSPOptions(options)
+	}
+	if len(goOvnOptions) != 0 {
+		cmd, err = ovnNBClient.LSPSetOptions(portName, goOvnOptions)
+		if err != nil {
+			return "", fmt.Errorf("failed to set options  %v to port %s error: %v", goOvnOptions, portName, err)
+		}
+		cmds = append(cmds, cmd)
+	}
+
+	err = ovnNBClient.Execute(cmds...)
+	if err != nil {
+		return "", fmt.Errorf("failed to add Logical Switch Portname %s  %v", portName, err)
+	}
+
+	//TODO: Replace once patch for LSPType gets merged in go-ovn
+	stdout, stderr, err := util.RunOVNNbctl("lsp-set-type", portName, portType)
 	if err != nil {
-		klog.Errorf("Failed to add logical port %s to switch %s, stdout: %q, stderr: %q, error: %v", portName, logicalSwitch, stdout, stderr, err)
+		klog.Errorf("Failed to set type %s for logical port %s to switch %s, stdout: %q, stderr: %q, error: %v", portType, portName, logicalSwitch, stdout, stderr, err)
 		return "", err
 	}
 
 	// UUID must be retrieved separately from the lsp-add transaction since
 	// (as of OVN 2.12) a bogus UUID is returned if they are part of the same
 	// transaction.
-	uuid, stderr, err := util.RunOVNNbctl("get", "logical_switch_port", portName, "_uuid")
+	lsp, err = ovnNBClient.LSPGet(portName)
 	if err != nil {
-		klog.Errorf("Error getting UUID for logical port %s "+
-			"stdout: %q, stderr: %q (%v)", portName, uuid, stderr, err)
+		klog.Errorf("Error getting UUID for logical port %s  %v lsp: %v", portName, err, lsp)
 		return "", err
 	}
-	if uuid == "" {
-		return uuid, fmt.Errorf("invalid logical port %s uuid", portName)
+
+	if lsp.UUID == "" {
+		return lsp.UUID, fmt.Errorf("invalid logical port %s uuid", portName)
+	}
+
+	return lsp.UUID, nil
+}
+
+//parses options in the form
```

Comment:
nit space after //

Reply:
the function doctext can just be all on one line as:

// parses options in the form "router-port=stor-node1" into a map